File: SofiForce.Models/Models/SearchModels/RepresentativeSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class RepresentativeSearchModel : BaseSearchModel
    {

        public int SupervisorId { get; set; }
        public int BranchId { get; set; }

        public int KindId { get; set; }
        public string KindIds { get; set; }



        public string RepresentativeName { get; set; }
        public DateTime? TerminationDate { get; set; }
        public DateTime? JoinDate { get; set; }
        public string Phone { get; set; }
        public int IsActive { get; set; }
        public int IsTerminated { get; set; }
        public int TerminationReasonId { get; set; }

        public int BusinessUnitId { get; set; }

        public string CompanyCode { get; set; }

    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/RepresentativeSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface RepresentativeSearchModel extends BaseSearchModel {
    supervisorId: number;
    supervisorCode:string;
    branchId: number;
    branchCode: string;

    kindId: number;
    kindIds: string;
    representativeName: string;
    terminationDate: Date | null;
    joinDate: Date | null;
    phone: string;
    isActive: number;
    isTerminated: number;
    terminationReasonId: number;
    businessUnitId: number;
    companyCode: string;
    searchMode:number;
}